.github/github_workflow_scripts/purge_branch_protection_rules.py

```diff
@@ -0,0 +1,358 @@
+"""
+A CLI to manage GitHub branch protection rules.
+
+It uses GitHub GraphQL API to query and remove
+branch protection rules.
+"""
+
+from dataclasses import dataclass
+import logging
+import os
+from pathlib import Path
+import sys
+from typing import Any
+
+from github import Github
+import github
+import github.Requester
+
+
+DEFAULT_REPO = "demisto/content"
+
+GH_TOKEN_ENV_VAR = "GITHUB_TOKEN"
+
+# https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+# e.g. 'demisto/content'
+GH_REPO_ENV_VAR = "GITHUB_REPOSITORY"
+GH_JOB_SUMMARY_ENV_VAR = "GITHUB_STEP_SUMMARY"
+PROTECTED_RULES = ["contrib/**/*"]
+
+# Logging setup
+LOG_FORMAT = "%(asctime)s %(levelname)s %(message)s"
+# Create logger
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)  # Set the lowest level to capture all messages
+
+# Create FileHandler and set level to DEBUG
+file_handler = logging.FileHandler(f"{Path(__file__).stem}.log")
+file_handler.setLevel(logging.DEBUG)
+file_handler.setFormatter(logging.Formatter(LOG_FORMAT))
+
+# Create StreamHandler and set level to INFO
+stream_handler = logging.StreamHandler()
+stream_handler.setLevel(logging.INFO)
+stream_handler.setFormatter(logging.Formatter(LOG_FORMAT))
+
+# Add handlers to the logger
+logger.addHandler(file_handler)
+logger.addHandler(stream_handler)
+
+
+@dataclass
+class BranchProtectionRule:
+    id: str
+    pattern: str
+    matching_refs: int
+
+
+# Queries
+GET_BRANCH_PROTECTION_GRAPHQL_QUERY_TEMPLATE = """query($owner: String!, $name: String!) {
+repository(owner: $owner, name: $name) {
+    branchProtectionRules(first: 100) {
+    nodes {
+        id
+        pattern
+        matchingRefs(first: 10) {
+        totalCount
+        }
+    }
+    }
+}
+}"""
+
+DELETE_BRANCH_PROTECTION_RULE_QUERY_TEMPLATE = """mutation deleteBranchProtectionRule($branchProtectionRuleId: ID!) {
+    deleteBranchProtectionRule(input: {branchProtectionRuleId: $branchProtectionRuleId}) {
+        clientMutationId
+    }
+}"""
+
+
+# Helper Functions
+
+def get_repo_owner_and_name() -> tuple[str, str]:
+    """
+    Extracts the repository owner and name from a given repository string.
+
+    Returns:
+        A `Tuple[str, str]` containing the repository owner and name.
+
+    Raises:
+        `ValueError`: If the input string is not in the expected 'owner/repository' format.
+    """
+
+    repo = os.getenv(GH_REPO_ENV_VAR, DEFAULT_REPO)
+
+    parts = repo.split('/')
+
+    if len(parts) != 2:
```

Comment:
what if the contributor's repo name has `/` in it? is it possible?

Reply:
Looks like the `/` is not a valid character, it will be replaced with `-`:

<img width="709" alt="image" src="https://github.com/user-attachments/assets/3e422bf9-1dfa-41c5-b111-5270e36989ed">